### PR TITLE
examples/imagimob: Adding imagimob mp integration.

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
### Summary

Integration for imagimob and micropython.

Knowing this is the first version, it does the job fine for now. I have only tested to check if imports are working fine. Praveen is going to eventually use/extend this for the application he is working on.

I suggest we don't merge this to ports-psoc6-main and rather keep the changes alive in imai-mp-integration branch for now.
In case we see the need to extend this, then:
1. We should explore how to add right architecture and then cross compile for our port.
2. Provide script that can be run from imagimob studio so that the workflow is fully automated.
3. Provide necessary docs.

For reviewing, you can ignore the non-mpy files.

### Testing

Details available in readme.md.

### CODE REVIEW

File: examples/usercimaimodule/imagimob/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Follow the steps below to generate micropython bindings for your imagimob generated model.
+
+1. The imagimob micropytthon integration is available in imai-mp-integration, so checkout to that branch after cloning this repository.
+
+        git clone https://github.com/Infineon/micropython.git
+
+        git checkout --track origin/ports-psoc6-main
+
+2. Then initialize the ModusToolbox™ environment:
```
Comment (jaenrig-ifx) [inline examples/usercimaimodule/imagimob/README.md:30]: You need to be already in ports/psoc6 to run this make target :)

**Review by jaenrig-ifx — APPROVED** [review]
All good, beside the small comment! 🦾

File: examples/usercimaimodule/imagimob/README.md (hunk 1 above)
Comment (NikhitaR-IFX) [reply examples/usercimaimodule/imagimob/README.md:30]: Fixed!

File: ports/psoc6/mpconfigport.h [hunk 2]
```diff
@@ -220,6 +220,7 @@ extern void machine_deinit();
 
 #define MICROPY_LOGGER_DEBUG                    (0)
 
+
```
Comment (ramya-subramanyam) [inline ports/psoc6/mpconfigport.h:223]: Can this file be reverted as there are no changes here?

**Review by ramya-subramanyam — APPROVED** [review]
looks good!

**Review by IFX-Anusha — APPROVED** [review]
Nice!

File: ports/psoc6/mpconfigport.h (hunk 2 above)
Comment (NikhitaR-IFX) [reply ports/psoc6/mpconfigport.h]: done

**Pull request merged**